forum/qa-include/pages/question-view.php

```diff
@@ -449,11 +449,21 @@ function qa_page_q_answer_view($question, $answer, $isselected, $usershtml, $for
 			$clicksuffix=' onclick="return qa_answer_click('.qa_js($answerid).', '.qa_js($question['postid']).', this);"';
 
 			if ($question['aselectable'] && !$answer['hidden'] && !$answer['queued']) {
-				if ($isselected)
+				if ($isselected){
 					$a_view['unselect_tags']='title="'.qa_lang_html('question/unselect_popup').'" name="'.$prefix.'dounselect"'.$clicksuffix;
-				else
-					$a_view['select_tags']='title="'.qa_lang_html('question/select_popup').'" name="'.$prefix.'doselect"'.$clicksuffix;
-			}
+				}else{
+					$query = 'SELECT * FROM qa_posts WHERE postid='.$answerid;

```

Comment:
- Czy czasem pod dostępną w tej funkcji zmienną `$answer` nie jest już cała odpowiedź? Jeśli tak, to po co niepotrzebnie pobierać to ponownie z bazy?
- A jeśli nawet już miałoby być pobierane z bazy, to nie należy używać do tabel wpisanego na stałe przedrostka `qa_` a znaku `^` - ten jest parsowany gdzieś przez funkcje Q2A i zamienia to na ten przedrostek, który u każdego może być inny (jest ustawiany bodajże w konfiguracji).
- Nie należy również podstawiać wartości bezpośrednio do zapytania, w Q2A jak pamiętam podaje się je jako kolejne argumenty i działa tam coś zabezpieczającego i walidującego.